gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcReadChannel.java

```diff
@@ -742,16 +742,28 @@ public SeekableByteChannel position(long newPosition) throws IOException {
     if (!isOpen()) {
       throw new ClosedChannelException();
     }
-    checkArgument(newPosition >= 0, "Read position must be non-negative, but was %s", newPosition);
-    checkArgument(
-        newPosition < size(),
-        "Read position must be before end of file (%s), but was %s",
-        size(),
-        newPosition);
+    validatePosition(newPosition, size());
     positionForNextRead = newPosition;
     return this;
   }
 
+  /** Validates that the given position is valid for this channel. */
+  protected void validatePosition(long position, long size) throws IOException {
+    if (position < 0) {
+      throw new EOFException(
```

Comment:
Should be throw IOException instead? @

Reply:
EOFException extends IOException only.

Reply:
Yes, but this is not really and EndOfFile case, correct? Why you think throwing an ArugmentException is not the right option here?

Reply:
Because that is a RuntimeException and clients are not expected to catch it.

Reply:
Can the position be negative, unless there is some bug? I could not think of a case where position can be negative, unless there is a bug and hence it is ok to crash in that case.

I will leave it to you to decide.